<launch>
  <arg name="map_name" default="map_1"/>
  <arg name="rviz" default="false"/>

  <!-- Preprocess Livox scans -->
  <rosparam command="load" file="$(find ptc_scan_processing)/config/params.yaml"/>
  <node name="preprocess_lidar_scan_node" pkg="ptc_scan_processing" type="preprocess_lidar_scan_node" output="screen"/>

  <!-- FAST LIO for lidar odometry -->
  <include file="$(find fast_lio)/launch/mapping_mid360.launch">
      <arg name="rviz" value="$(arg rviz)"/>
  </include>

  <!-- Load the YAML file with parameters -->
  <rosparam command="load" file="$(find mapping)/config/params.yaml"/>
  <param name="map_data/save_relative_path" value="maps"/>
  <param name="map_data/map_name" value="$(arg map_name)"/>
  <rosparam command="load" file="$(find vehicle_params)/config/params.yaml"/>

  <!-- Launch the mapping node -->
  <node pkg="mapping" type="map_data_save_node" name="mapping_node" output="screen"/>
</launch>
